### ISSUE TITLE
[Bug] ArgoCD compatibility

### ISSUE THREAD

**Issue by grzesuav** [cryostatio/cryostat-helm#187]
### Current Behavior

As for now, when deploying with ArgoCD make the app desyncronized -  see picture.
<img width="1112" alt="image" src="https://github.com/user-attachments/assets/91550789-ead5-4dad-ba1c-18b123cf14b5">

### Expected Behavior

No sync differences 

### Steps To Reproduce

Just deploy cryostat via argocd

### Environment

```Markdown
- OS: 
- Environment: 
- Version:
```

### Anything else?

I suspect this is because of some helm random function used to generate one time password

**Comment by grzesuav** [comment]
Main problem is `lookup` helm function is not supported in argocd - https://github.com/cryostatio/cryostat-helm/blob/main/charts/cryostat/templates/_helpers.tpl#L104
https://github.com/argoproj/argo-cd/issues/5202

would be nice to set this values manually (i.e.have possibility to define secret name in values, similar like https://github.com/cryostatio/cryostat-helm/blob/main/charts/cryostat/templates/db_secret.yaml#L1 is handled)

**Comment by grzesuav** [comment]
the same I guess with https://github.com/cryostatio/cryostat-helm/blob/main/charts/cryostat/templates/cookie_secret.yaml#L1

**Comment by andrewazores** [comment]
I filed #188 above as a separate issue to track the feature of supplying all the auto-generated secret values manually - it's a good idea in general. If that ends up being everything required to fix ArgoCD compatibility then great, otherwise I leave this as a separate issue in case more is required after that work is done.

**Comment by andrewazores** [comment]
Linking to a relevant comment from the discussion above: https://github.com/argoproj/argo-cd/issues/5202#issuecomment-834134792

This comment is describing exactly the same thing we do with the same motivation.

Other useful-looking links from that discussion:

- https://github.com/kuuji/helm-external-val
- https://github.com/jkroepke/helm-kubectl

----

On the other hand, from that same discussion, ArgoCD is trying to come up with a solution so that Helm charts like ours should be usable (ie let the `lookup` function work).

https://github.com/argoproj/argo-cd/issues/5202#issuecomment-1435383896

https://github.com/argoproj/argo-cd/issues/5202#issuecomment-1644432182

So there may be some solution on the Argo side as well, even if we don't get around to fully fixing this in our Helm chart first.

**Comment by andrewazores** [comment]
@grzesuav could you try deploying https://github.com/cryostatio/cryostat-helm/pull/189 in ArgoCD and see if your sync issue is resolved? You will need to create the three secrets manually first and then supply the name of each one to the Helm chart when you install it.

**Comment by grzesuav** [comment]
@andrewazores is the helm chart available somewhere so I can pick it up ?

Regarding argocd, the issue is been there for like 2/3 years so I do not expect it will land soon.

**Comment by andrewazores** [comment]
No, the only place to test that change would be the PR itself - you'd have to clone the repository and install it manually.

**Comment by grzesuav** [comment]
just getting back to it, I see pr is merged, is it part of theofficial release ?

**Comment by andrewazores** [comment]
It will be, the next time there is a release.

**Comment by andrewazores** [comment]
@grzesuav just looping back on this - we're in the midst of releasing 4.0 right now. The downstream product build is underway and there is some paperwork etc. to complete, but the upstream release is pretty much done at this point and includes #189.

**Comment by andrewazores** [comment]
Whoops, didn't mean to close, just to comment.
